python/boardgameio.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2018 The boardgame.io Authors
+#
+# Use of this source code is governed by a MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+
+import logging, time
+import socketIO_client as io
+
+
+class Namespace(io.BaseNamespace):
+
+    log = logging.getLogger('client.namespace')
+    
+    def __init__(self, *args):
+        io.BaseNamespace.__init__(self, *args)
+        self.bot = None
+        self.previous_state_id = None
+        self.actions = []
+
+    def set_bot_info(self, bot):
+        self.bot = bot
+    
+    def on_connect(self):
+        self.log.info('connected') # to game <%s>' % self.bot.game_id)
+    def on_disconnect(self):
+        self.log.info('disconnected')
+    def on_reconnect(self):
+        self.log.info('reconnected')
+    
+    def on_sync(self, *args):
+        game_id = args[0]
+        state = args[1]
+        state_id = state['_stateID']
+        ctx = state['ctx']
+        player = ctx['actionPlayers'][0]
```

Comment:
The semantics of `actionPlayers` is that any player in the set is allowed to play. So you should be checking if:

self.bot.player_id in actionPlayers

instead of just looking at the first value